reconcile/external_resources/manager.py

```diff
@@ -92,6 +92,7 @@ class ReconcileAction(StrEnum):
 class ReconciliationStatus(BaseModel):
     reconcile_time: int = 0
     resource_status: ResourceStatus
+    needs_state_update: bool = False
```

Comment:
Please make it a `property` as we have for [ResourceStatus](https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/external_resources/state.py#L46)

Reply:
Do you mean using logic depending the ResourceStatus values ?

Reply:
Yes, exactly

Reply:
The logic to update this attribute depends on the state's resource_status and the reconciliation resource_status. It can not be set only with its resource_status attribute. 

passing the state's to a method to set this as a property feels a bit weird to me.

Reply:
I don't see the dependency on the state's `resource_status`. For me, it looks like

```
@propery
def needs_state_update(self) -> bool:
    return self.resource_state in [ResourceStatus.DELETED, ResourceStatus.ERROR]
```

Reply:
needs_state_update is False by default, it's only set True in `_get_reconciliation_status`  if the state.resource_status is in progress.

Reply:
https://github.com/app-sre/qontract-reconcile/pull/4775#discussion_r1856614330 can avoid that field or property